### PR TITLE
Making Seat optional on course-creation page.

### PR DESCRIPTION
**Pull request by awais786**
Seat type and prices are marked as required=False in forms.
Hide the div in page using waffle switch. 
Add data migration to create the switch.

ECOM-6797

### CODE REVIEW

**Comment by awais786** [comment]
@tasawernawaz Please review.

File: course_discovery/apps/publisher/forms.py [hunk 1]
```diff
@@ -250,7 +250,8 @@ def __init__(self, *args, **kwargs):
         (Seat.PROFESSIONAL, _('Professional Education')),
     ]
 
-    type = forms.ChoiceField(choices=TYPE_CHOICES, required=True, label=_('Seat Type'))
+    type = forms.ChoiceField(choices=TYPE_CHOICES, required=False, label=_('Seat Type'))
+    price = forms.DecimalField(max_digits=5, decimal_places=2, required=False, initial=0.00)
```
Comment (bderusha) [inline course_discovery/apps/publisher/forms.py]: lets make the max 6 digits

File: course_discovery/apps/publisher/migrations/0026_create_switch_hide_features_for_pilot.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+from django.db import migrations
+
+
+def create_switch(apps, schema_editor):
+    """Create the enable_publisher_email_notifications switch if it does not already exist."""
```
Comment (bderusha) [inline course_discovery/apps/publisher/migrations/0026_create_switch_hide_features_for_pilot.py]: update doc string

File: course_discovery/apps/publisher/migrations/0026_create_switch_hide_features_for_pilot.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    Switch = apps.get_model('waffle', 'Switch')
+    Switch.objects.get_or_create(name='publisher_hide_features_for_pilot', defaults={'active': False})
+
+
+def delete_switch(apps, schema_editor):
+    """Delete the enable_publisher_email_notifications switch."""
```
Comment (bderusha) [inline course_discovery/apps/publisher/migrations/0026_create_switch_hide_features_for_pilot.py]: update docstring

**Review by bderusha — APPROVED** [review]
A couple small things then 👍

**Comment by awais786** [comment]
@tasawernawaz @waheedahmed Please review.

File: course_discovery/templates/publisher/add_course_form.html [hunk 4]
```diff
@@ -164,7 +164,7 @@ <h2 class="layout-title">{% trans "Studio instance information" %}</h2>
 </div>
 
 
-<div class="layout-full publisher-layout layout">
+<div class="layout-full publisher-layout layout" style="{% if publisher_hide_features_for_pilot %}display:none;{% endif %}">
```
Comment (waheedahmed) [inline course_discovery/templates/publisher/add_course_form.html]: Instead adding style here you can simply add ```hidden``` class.

File: course_discovery/apps/publisher/tests/test_views.py [hunk 5]
```diff
@@ -108,6 +109,20 @@ def test_create_course_and_course_run_and_seat_with_errors(self):
...
+        """ Verify that course and course run objects create successfully if seat type
+        is not provided.
+        """
+        data = {'number': 'course_without_seat', 'image': ''}
+        course_dict = self._post_data(data, self.course, self.course_run, None)
+        course_dict['image'] = ''
+        self.client.post(reverse('publisher:publisher_courses_new'), course_dict)
+
+        course = Course.objects.get(number=course_dict['number'])
+        course_run = course.publisher_course_runs.all()[0]
```
Comment (waheedahmed) [inline course_discovery/apps/publisher/tests/test_views.py]: Instead indexing use ```course.publisher_course_runs.first()```.

**Review by waheedahmed — CHANGES_REQUESTED** [review]

File: course_discovery/templates/publisher/add_course_form.html (hunk 4 above)
Comment (awais786) [reply course_discovery/templates/publisher/add_course_form.html]: Do you mean add a hidden class in sass ?

Comment (waheedahmed) [reply course_discovery/templates/publisher/add_course_form.html]: Its already available just add it here in class attribute.

**Review by waheedahmed — APPROVED** [review]

**Pull request merged**